frontend/src/components/Header.tsx

```diff
@@ -119,11 +119,40 @@ const NavLinks = ({
   );
 };
 
+const LoginLink = ({ navLoginLinkText }: { navLoginLinkText: string }) => {
+  const [authLoginUrl, setAuthLoginUrl] = useState<string | null>(null);
```

Comment:
no need to make a change, but for future reference this could be simplified since useState should always accept an undefined initial value
```
const [authLoginUrl, setAuthLoginUrl] = useState<string>();
```